mindmeld/components/question_answerer.py

```diff
@@ -454,106 +708,189 @@ class Indices:
+            disk for fast loading time later on. Note that this method is best used with the
+            Resolver.load_kb() method wherein fit resources are created.
+
+            During reloading of an index, use self.get_index_metadata() as well as
+            fieldResource.update_resource() to obtain back a fit index.
 
-            # update
+            Args:
+                index_name (str): A scoped index name for loading metadata
+                index_resources (Dict[str, FieldResource]): Dict curated with all the KB fields and
+                    their FieldResources
+                 index_all_ids (List[str]): List of all ids observed for this index in the order
+                    they are present in the KB.
+            """
+
+            # update memory
             self._indices.update({index_name: index_resources})
             self._indices_all_ids.update({index_name: index_all_ids})
 
-            # create repo for dumping if required
+            # dump to disk
             cache_path = self._get_index_cache_path(index_name)
-            dir_name = os.path.dirname(cache_path)
-            if not os.path.exists(dir_name):
-                os.makedirs(dir_name)
-
-            # dump
-            metadata_dump = self._create_metadata_for_dumping(index_name)
-            metadata_dump.update({"__all_ids": index_all_ids})
-            opfile = open(cache_path, "wb")
-            pickle.dump(metadata_dump, opfile)
-            opfile.close()
+            os.makedirs(os.path.dirname(cache_path), exist_ok=True)
+            metadata, index_all_ids = self._get_index_metadata_from_memory(index_name)
+            metadata.update({"__all_ids": index_all_ids})
+            with open(cache_path, "wb") as opfile:
+                pickle.dump(metadata, opfile)
+                opfile.close()
 
-        def is_available(self, index_name):
-            return index_name in self or os.path.exists(self._get_index_cache_path(index_name))
+        def readify_index(self, index_name):
+            """
+            Makes the index resource ready to be used for inference. If the index is in memory, it
+            means that the index resources are already ready to be used for inference. If not, this
+            method loads metadata, fits resources and gets them ready.
+            """
+            if index_name in self:
+                # in-memory indices are always fit indices
+                return
+            elif self.is_available(index_name):
+                index_resources, index_all_ids = self.get_index_metadata(index_name)
+                # Instantiate one resource loader for all resolvers being loaded in Native QA
+                _resource_loader = NativeQuestionAnswerer.get_resource_loader()
+                for field_name, field_resource in index_resources.items():
+                    field_resource.update_resource(
+                        id2value={},
```

Comment:
why empty dict here?

Reply:
(Related [comment]https://github.com/cisco/mindmeld/pull/341#discussion_r720574972) 

So by passing id2value as en empty dict, we can trigger loading of resolvers. But because it doesn't make a good code style, I now modified the code to have two separate methods- update_resource and load_resource, the latter taking care of only loading resolvers for FieldResources. So this line 877 that you pointed no longer exists.

@mhuang2173 @vijay120